core/src/services/alluxio/backend.rs

```diff
@@ -108,12 +108,9 @@ impl Builder for AlluxioBuilder {
         let root = normalize_root(&self.config.root.clone().unwrap_or_default());
         debug!("backend use root {}", &root);
 
-        let endpoint = match &self.config.endpoint {
-            Some(endpoint) => Ok(endpoint.clone()),
-            None => Err(Error::new(ErrorKind::ConfigInvalid, "endpoint is empty")
-                .with_operation("Builder::build")
-                .with_context("service", Scheme::Alluxio)),
-        }?;
+        let endpoint =
+            Error::ensure_endpoint_not_empty(self.config.endpoint.as_deref(), Self::SCHEME)
```

Comment:
Thanks a lot for the PR, but I don't want to optimize code in this way.

Reply:
The reason is similiar to https://github.com/apache/opendal/pull/5037#issuecomment-2308400902